docs/source/integrations/mlflow.md

```diff
@@ -0,0 +1,278 @@
+
+## Use cases
+
+There are different use cases for which you might want to use MLflow,
+some of which are already covered by the [`kedro-mlflow`](https://kedro-mlflow.readthedocs.io/) plugin.
+This page explains how to implement them yourself in a simplified form,
+and also how `kedro-mlflow` can do the hard work for you.
+
+### Simple tracking of Kedro runs in MLflow using Hooks
+
+Although MLflow works best when working with machine learning (ML) and AI pipelines,
+you can track your regular Kedro runs as experiments in MLflow even if they do not use ML.
+
+One possible way of doing it is using the {py:meth}`~kedro.framework.hooks.specs.PipelineSpecs.before_pipeline_run` Hook
+to log the `run_params` passed to the Hook.
+An implementation would look as follows:
+
+```python
+import typing as t
+
+import mlflow
+from kedro.framework.hooks import hook_impl
+from mlflow.entities import RunStatus
+
+
+class MLflowHooks:
+    @hook_impl
+    def before_pipeline_run(self, run_params: dict[str, t.Any]):
+        logger.info("Logging run parameters to MLflow")
+        mlflow.log_params(run_params)
+        mlflow.set_tags({"pipeline": run_params["pipeline_name"] or "__default__"})
+
+    @hook_impl
+    def on_pipeline_error(self):
+        logger.error("Logging error to MLflow")
+        mlflow.end_run(RunStatus.to_string(RunStatus.FAILED))
+```
+
+Notice that there is also extra code to mark the run as failed if there is any error.
+
+After enabling this custom Hook, you can execute `kedro run`, and see something like this in the logs:
+
+```
+[05/06/24 12:43:30] INFO     Kedro project spaceflights-mlflow             session.py:324
+[05/06/24 12:43:31] INFO     Logging run parameters to MLFlow                 hooks.py:18
+...
+```
+
+If you open your tracking server UI you will observe a result like this:
+
+```{image} ../meta/images/simple-mlflow-tracking1.png
+:alt: Simple MLflow tracking
+:width: 50%
+:align: center
+```
+
+```{image} ../meta/images/simple-mlflow-tracking2.png
+:alt: Simple MLflow tracking
+:width: 50%
+:align: center
+```
+
+Notice that these "parameters" do not include the Kedro parameters as defined in `parameters*.yml`.
+To log the actual Kedro parameters, you can continue writing a custom Hook
+or use the `kedro-mlflow` plugin instead, as described in the following section.
+
+### Complete tracking of Kedro runs in MLflow using `kedro-mlflow`
+
+You could make the code snippet above more complex, but you are likely to reach a point where it is easier to take advantage of the work that the `kedro-mlflow` plugin can do for you.
+
+To start using `kedro-mlflow`, the only required step is to install it:
+
+```
+pip install kedro-mlflow
+```
+
+In recent versions of Kedro, this will already register the `kedro-mlflow` Hooks for you.
+
+:::{note}
+If you implemented your custom `MLflowHooks` as described in the previous section,
+consider disabling them to avoid logging the runs twice.
+:::
```

Comment:
I think this is confusing for beginners and the warning should be at the beginning of the section. If one's read the section from linearly top to bottom (which is what you usually do 😄) I think it is very confusing to understand that this section suggests to *replace* the custom hook with ``kedro-mlflow`` and not add it.  Eventually as a beginner I would likely wonder "why the hell did I start to do something that is suggested t be removed in the following section? 

I think an (advanced) example that would be much nicer could be an example of a *combination* of ``kedro-mlflow`` and a custom hook (which will for exemple track something custom that ``kedro-mlflow`` does not, e.g. the commit sha ?), but still benefit from kedro-mlflow configuration management because kedro-mlflow hook will run first.